File: crates/psychevo/tests/fixtures/plugin_worker_tool_discovery.py
#!/usr/bin/env python3
import json, sys
for line in sys.stdin:
    req=json.loads(line)
    method=req.get("method")
    if method=="initialize":
        result={"ok": True}
    elif method=="contributions/list":
        result={"tools":[{"name":"cleanup_status","description":"status","parameters":{"type":"object","properties":{}}}]}
    else:
        result={}
    print(json.dumps({"jsonrpc":"2.0","id":req.get("id"),"result":result}), flush=True)

File: crates/psychevo/tests/fixtures/plugin_worker_tool_discovery.js
#!/usr/bin/env node
const readline = require("readline");

readline.createInterface({ input: process.stdin, crlfDelay: Infinity }).on("line", (line) => {
  const request = JSON.parse(line);
  let result = {};
  if (request.method === "initialize") {
    result = { ok: true };
  } else if (request.method === "contributions/list") {
    result = {
      tools: [{ name: "cleanup_status", description: "status", parameters: { type: "object", properties: {} } }],
    };
  }
  process.stdout.write(`${JSON.stringify({ jsonrpc: "2.0", id: request.id, result })}\n`);
});
